server/app/models/distributed_lock.rb

```diff
@@ -13,17 +15,16 @@ def self.with_lock(name, timeout = 60)
     lock_id = nil
     begin
       if timeout.to_f > 0.0
-        Timeout.timeout(timeout) do
-          sleep 0.01 until lock_id = self.obtain_lock(name)
-        end
+        lock_id = WaitHelper.wait(timeout: timeout) { self.obtain_lock(name) }
```

Comment:
Because this is a class method, so just `include WaitHelper` doesn't let you call `self.wait`

```
  1) DistributedLock.with_lock returns false if getting lock timeouts
     Failure/Error: described_class.with_lock('foo1') { sleep 0.1 }
     NoMethodError:
       undefined method `wait' for DistributedLock:Class
       Did you mean?  with
     # ./app/models/distributed_lock.rb:19:in `with_lock'
     # ./spec/models/distributed_lock_spec.rb:27:in `block (4 levels) in <top (required)>'
```

Is there a better way to get this working?